lib/bibliothecary/parsers/pypi.rb

```diff
@@ -144,17 +144,25 @@ def self.map_dependencies(packages, type, source = nil)
             # package. Break these out into a separate dep per requirement.
             # https://python-poetry.org/docs/dependency-specification/#multiple-constraints-dependencies
             package_info.map do |info|
+              # Poetry normalizes names in lockfiles but doesn't provide the original, so we need to keep
+              # track of the original name so the dep is connected between manifest+lockfile.
+              normalized_name = normalize_name(name)
               Dependency.new(
-                name: name,
+                name: normalize_name ? normalized_name : name,
+                original_name: normalize_name && name != normalized_name ? name : nil,
                 requirement: map_requirements(info),
                 type: type,
                 source: source,
                 local: local
               )
             end
           else
+            # Poetry normalizes names in lockfiles but doesn't provide the original, so we need to keep
+            # track of the original name so the dep is connected between manifest+lockfile.
+            normalized_name = normalize_name(name)
             Dependency.new(
-              name: name,
+              name: normalize_name ? normalized_name : name,
```

Comment:
Shouldn't this be
```suggestion
              name: normalize_named ? normalized_name : name,
```

Reply:
(the `normalized_name` computation on line 162 is just a way to DRY it up so we don't have to call `normalize_name()` twice too)